plugins/polio/api/vaccines/stock_management.py

```diff
@@ -78,11 +75,169 @@ class VaccineStockManagementReadWritePerm(GenericReadWritePerm):
+            results.append(
+                {
+                    "date": report.destruction_report_date,
+                    "action": f"{report.action} ({report.lot_number})"
+                    if len(report.action) > 0
+                    else f"Stock Destruction ({report.lot_number})",
+                    "vials_in": None,
+                    "doses_in": None,
+                    "vials_out": report.unusable_vials_destroyed,
+                    "doses_out": "N/A",
+                }
+            )
+
+        for report in incident_reports:
+            results.append(
+                {
+                    "date": report.date_of_incident_report,
+                    "action": report.stock_correction,
+                    "vials_in": None,
+                    "doses_in": None,
+                    "vials_out": report.unusable_vials + report.usable_vials,
+                    "doses_out": "N/A",
+                }
+            )
+
+        for movement in stock_movements:
+            if movement.usable_vials_used > 0:
+                results.append(
+                    {
+                        "date": movement.report_date,
+                        "action": "Form A - Vials Used",
+                        "vials_in": None,
+                        "doses_in": None,
+                        "vials_out": movement.usable_vials_used,
+                        "doses_out": "N/A",  # can't compute with current model
+                    }
+                )
+            elif movement.unusable_vials > 0:
+                results.append(
+                    {
+                        "date": movement.report_date,
+                        "action": "Form A - Unusable Vials",
+                        "vials_in": None,
+                        "doses_in": None,
+                        "vials_out": movement.unusable_vials,
+                        "doses_out": "N/A",  # can't compute with current model
+                    }
+                )
+            elif movement.missing_vials > 0:
+                results.append(
+                    {
+                        "date": movement.report_date,
+                        "action": "Form A - Missing Vials",
+                        "vials_in": None,
+                        "doses_in": None,
+                        "vials_out": movement.missing_vials,
+                        "doses_out": "N/A",  # can't compute with current model
+                    }
+                )
+
+        paginator = Paginator()
+        page = paginator.paginate_queryset(sorted(results, key=lambda x: x["date"]), request)
+        if page is not None:
+            return paginator.get_paginated_response(page)
+        return Response({"results": results})
+
+    @action(detail=True, methods=["get"])
+    def get_unusable_vials(self, request, pk=None):
+        pass
+
+    @action(detail=True, methods=["get"])
+    def get_summary(self, request, pk=None):
+        pass
```

Comment:
These views (`get_unusable_vials` and `get_summary`) are literally doing nothing.

Why are we keeping them?

Reply:
I'm in the process of writing them for next PR